ui/packages/catalog-ui-search/src/main/webapp/react-component/location/location.js

```diff
@@ -103,17 +145,63 @@ const ddValidators = {
   east: value => value <= 180 && value >= -180,
 }
 
+let isDms = false
+const dmsValidators = {
+  dmsLat: value => validateInput(value, 'dd°mm\'ss.s"'),
+  dmsLon: value => validateInput(value, 'ddd°mm\'ss.s"'),
+  dmsNorth: value => validateInput(value, 'dd°mm\'ss.s"'),
+  dmsSouth: value => validateInput(value, 'dd°mm\'ss.s"'),
+  dmsWest: value => validateInput(value, 'ddd°mm\'ss.s"'),
+  dmsEast: value => validateInput(value, 'ddd°mm\'ss.s"'),
+}
+
+const getNegOrPosLatLon = (key, value) => {
+  if (value < 0) {
+    return -1 * validLatLon[key]
+  } else {
+    return validLatLon[key]
+  }
+}
+
 module.exports = ({ state, setState, options }) => (
   <LocationInput
     {...state}
     onDraw={options.onDraw}
     setState={setState}
     cursor={key => value => {
```

Comment:
✏️ Would be nice to move this function definition outside of the component definition now that it has grown a fair bit